src/kaleidoscope_internal/deprecations.h

```diff
@@ -51,5 +51,17 @@
   "`Kaleidoscope.device().led_count` instead."
 
 #define _DEPRECATED_MESSAGE_HARDWARE_BASE_CLASS                 \
-  "The `Hardware` base class is deprecated. Please use\n" __NL_ \
+  "The `Hardware` base class is deprecated. Please use\n" __NL__ \
   "the new APIs based on `kaleidoscope::device::Base`."
+
+#define _DEPRECATED_MESSAGE_DIRECT_KEY_MEMBER_ACCESS                           \
+  "Direct access to `Key` class' data members is deprecated.\n"                \
+  "Please use `Key::setKeyCode()`/`Key::getKeyCode()` or\n"                    \
+  "`Key::setFlags()`/`Key::getFlags()` instead."
```

Comment:
I think this should be reworded a bit. It's clear to anyone who knows C++, but for someone who's just copy & pasting code, this might as well be Klingon. I think a better way would be to point them to `UPGRADING.md`, where we can provide a few examples of how to get from the old style to the new one.

To be honest, just slap `For further information and examples on how to do that, please see UPGRADING.md`.

Reply:
Modified `deprecations.h`, now pointing to `UPGRADING.md`.

Reply:
Now, as even Duolingo supports Klingon, one can expect more and more Klingon speaking users around :smirk: